plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,454 @@
+	fields["upstream_curr_rate"] = info.NewUpstreamCurrRate
+	fields["downstream_curr_rate"] = info.NewDownstreamCurrRate
+	fields["upstream_max_rate"] = info.NewUpstreamMaxRate
+	fields["downstream_max_rate"] = info.NewDownstreamMaxRate
+	fields["upstream_noise_margin"] = info.NewUpstreamNoiseMargin
+	fields["downstream_noise_margin"] = info.NewDownstreamNoiseMargin
+	fields["upstream_attenuation"] = info.NewUpstreamAttenuation
+	fields["downstream_attenuation"] = info.NewDownstreamAttenuation
+	fields["upstream_power"] = info.NewUpstreamPower
+	fields["downstream_power"] = info.NewDownstreamPower
+	fields["receive_blocks"] = statisticsTotal.NewReceiveBlocks
+	fields["transmit_blocks"] = statisticsTotal.NewTransmitBlocks
+	fields["cell_delin"] = statisticsTotal.NewCellDelin
+	fields["link_retrain"] = statisticsTotal.NewLinkRetrain
+	fields["init_errors"] = statisticsTotal.NewInitErrors
+	fields["init_timeouts"] = statisticsTotal.NewInitTimeouts
+	fields["loss_of_framing"] = statisticsTotal.NewLossOfFraming
+	fields["errored_secs"] = statisticsTotal.NewErroredSecs
+	fields["severly_errored_secs"] = statisticsTotal.NewSeverelyErroredSecs
+	fields["fec_errors"] = statisticsTotal.NewFECErrors
+	fields["atuc_fec_errors"] = statisticsTotal.NewATUCFECErrors
+	fields["hec_errors"] = statisticsTotal.NewHECErrors
+	fields["atuc_hec_errors"] = statisticsTotal.NewATUCHECErrors
+	fields["crc_errors"] = statisticsTotal.NewCRCErrors
+	fields["atuc_crc_errors"] = statisticsTotal.NewATUCCRCErrors
+	acc.AddFields("fritzbox_dsl", fields, tags)
+	if plugin.Log.Level().Includes(telegraf.Debug) {
+		plugin.logMetric("fritzbox_dsl", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) gatherWlanInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	serviceClient := wlanconfig.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	info := &wlanconfig.GetInfoResponse{}
+	err := serviceClient.GetInfo(info)
+	if err != nil {
+		return err
+	}
+	if info.NewStatus != "Up" {
+		return nil
+	}
+	totalAssociations := &wlanconfig.GetTotalAssociationsResponse{}
+	err = serviceClient.GetTotalAssociations(totalAssociations)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]string)
+	tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+	tags["service"] = serviceClient.Service.ShortId()
+	tags["ssid"] = info.NewSSID
+	tags["channel"] = strconv.Itoa(int(info.NewChannel))
+	tags["band"] = plugin.wlanBandFromInfo(info)
+	fields := make(map[string]interface{})
+	fields["total_associations"] = totalAssociations.NewTotalAssociations
+	acc.AddGauge("fritzbox_wlan", fields, tags)
+	if plugin.Log.Level().Includes(telegraf.Debug) {
+		plugin.logMetric("fritzbox_wlan", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) wlanBandFromInfo(info *wlanconfig.GetInfoResponse) string {
+	band := info.NewX_AVM_DE_FrequencyBand
+	if band != "" {
+		return band
+	}
```

Comment:
Does it ever happen info.NewX_AVM_DE_FrequencyBand is empty?

Reply:
Yes, this property has only been introduced with the latest firmware (8.0). For older devices this is empty.